### ISSUE TITLE
Strange behaviour

### ISSUE THREAD

**Issue by bra-fsn** [jnwatson/py-lmdb#50]
Using git master and lmdb 0.9.10 on FreeBSD with the following simple writer:

``` python
import lmdb
import time

env = lmdb.open('pb',map_size=1024**4,metasync=False,sync=False)

recnum=1000000
data='A'*250
start=time.time()
for i in xrange(recnum):
        with env.begin(write=True) as txn:
                txn.put(str(i), data)
end=time.time()
print end-start, recnum/(end-start)
env.close()
```

the process starts with 5896K memory usage and ends with 130M.
Re-running the same program again yields 408M memory usage (145M just after opening the environment). Running it third times it also gives a 408M maximum.
Why is this?

Also, I have a "reader", which is supposed to read all the records and delete them if processed:

``` python
import lmdb

env = lmdb.open('pb',map_size=1024**4,metasync=False,sync=False)

while True:
        sleep=False
        with env.begin(write=True) as txn:
                cur=txn.cursor()
                res=cur.first()
                if res:
                        cur.delete()
                else:
                        sleep=True
        if sleep:
                time.sleep(0.1)
```

This runs for few seconds and the dies:
lmdb.PageFullError: mdb_cursor_del: MDB_PAGE_FULL: Internal error - page has no more space

Basically I would like to do a two process (reader/writer) FIFO queue here (I'm aware that the sorting is alphabetical, so the order is not perfect).

I'm new to lmdb, so please bear with me, if I misunderstood something.
